### ISSUE TITLE
Segfault when trying to document Redshift database

### ISSUE THREAD

**Issue by craigrosie** [k1LoW/tbls#111]
```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x41f7cf6]

goroutine 1 [running]:
database/sql.(*Rows).close(0x0, 0x0, 0x0, 0x0, 0x0)
        /Users/k1low/.anyenv/envs/goenv/versions/1.12.5/src/database/sql/sql.go:2976 +0x66
database/sql.(*Rows).Close(0x0, 0x48c2a0c, 0x172)
        /Users/k1low/.anyenv/envs/goenv/versions/1.12.5/src/database/sql/sql.go:2972 +0x33
github.com/k1LoW/tbls/drivers/postgres.(*Postgres).Analyze(0xc0003397c0, 0xc0000dd4f0, 0x498d880, 0xc0003b6f40)
        /Users/k1low/src/github.com/k1LoW/tbls/drivers/postgres/postgres.go:131 +0x2f8a
github.com/k1LoW/tbls/datasource.Analyze(0x7ffeefbfe421, 0xa0, 0x0, 0x0, 0x0)
        /Users/k1low/src/github.com/k1LoW/tbls/datasource/datasource.go:74 +0x683
github.com/k1LoW/tbls/cmd.glob..func2(0x4e5dbc0, 0xc0001f7460, 0x1, 0x1)
        /Users/k1low/src/github.com/k1LoW/tbls/cmd/doc.go:72 +0x111
github.com/spf13/cobra.(*Command).execute(0x4e5dbc0, 0xc0001f7430, 0x1, 0x1, 0x4e5dbc0, 0xc0001f7430)
        /Users/k1low/go/pkg/mod/github.com/spf13/cobra@v0.0.4/command.go:766 +0x2ae
github.com/spf13/cobra.(*Command).ExecuteC(0x4e5d4a0, 0x0, 0x0, 0x45eabea)
        /Users/k1low/go/pkg/mod/github.com/spf13/cobra@v0.0.4/command.go:850 +0x2fc
github.com/spf13/cobra.(*Command).Execute(...)
        /Users/k1low/go/pkg/mod/github.com/spf13/cobra@v0.0.4/command.go:800
github.com/k1LoW/tbls/cmd.Execute()
        /Users/k1low/src/github.com/k1LoW/tbls/cmd/root.go:59 +0x2e
main.main()
        /Users/k1low/src/github.com/k1LoW/tbls/main.go:32 +0x20
```

```bash
$ tbls version
1.15.1
```

```bash
$ go version
go version go1.12.5 darwin/amd64
```

Please let me know if there's any other information I can provide to help debugging!

**Comment by k1LoW** [comment]
Hi @craigrosie ! Thank you for your report !!

Could you execute following query to your Amazon Redshift database ?

```sql
SELECT
i.relname AS indexname,
pg_get_indexdef(i.oid) AS indexdef
FROM ((((pg_index x
JOIN pg_class c ON ((c.oid = x.indrelid)))
JOIN pg_class i ON ((i.oid = x.indexrelid)))
LEFT JOIN pg_namespace n ON ((n.oid = c.relnamespace))))
WHERE ((c.relkind = ANY (ARRAY['r'::"char", 'm'::"char"])) AND (i.relkind = 'i'::"char"))
AND c.relname = 'your table name'
AND n.nspname = 'your database name'
ORDER BY x.indexrelid
```

**Comment by craigrosie** [comment]
I wasn't sure which table name I should use, so I put
```go
log.Println("Table name: " + tableName)
```
before https://github.com/k1LoW/tbls/blob/master/drivers/postgres/postgres.go#L131 which printed `2019/05/27 10:00:17 Table name: getstat`.

Running the query you suggested on that table gives:
```
ERROR:  cannot cast type smallint[] to character varying
```

**Comment by craigrosie** [comment]
@k1LoW Thanks for addressing this so quickly in #114!

I've pulled from `master` and rebuilt the `tbls` binary, and now I get a segfault in a different place (still on the same table):

```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x41f7a66]

goroutine 1 [running]:
database/sql.(*Rows).close(0x0, 0x0, 0x0, 0x0, 0x0)
        /usr/local/opt/go/libexec/src/database/sql/sql.go:2976 +0x66
database/sql.(*Rows).Close(0x0, 0x48c2002, 0x192)
        /usr/local/opt/go/libexec/src/database/sql/sql.go:2972 +0x33
github.com/k1LoW/tbls/drivers/postgres.(*Postgres).Analyze(0xc0001cbd00, 0xc00029ac80, 0x498c9c0, 0xc0003d8160)
        /Users/craigrosie/github/tbls/drivers/postgres/postgres.go:292 +0x2664
github.com/k1LoW/tbls/datasource.Analyze(0x7ffeefbfe483, 0xa0, 0x0, 0x0, 0x0)
        /Users/craigrosie/github/tbls/datasource/datasource.go:74 +0x683
github.com/k1LoW/tbls/cmd.glob..func2(0x4e5abc0, 0xc000298ef0, 0x1, 0x1)
        /Users/craigrosie/github/tbls/cmd/doc.go:72 +0x111
github.com/spf13/cobra.(*Command).execute(0x4e5abc0, 0xc000298ec0, 0x1, 0x1, 0x4e5abc0, 0xc000298ec0)
        /Users/craigrosie/github/golang/pkg/mod/github.com/spf13/cobra@v0.0.4/command.go:766 +0x2ae
github.com/spf13/cobra.(*Command).ExecuteC(0x4e5a4a0, 0x0, 0x0, 0x45ea0ba)
        /Users/craigrosie/github/golang/pkg/mod/github.com/spf13/cobra@v0.0.4/command.go:850 +0x2fc
github.com/spf13/cobra.(*Command).Execute(...)
        /Users/craigrosie/github/golang/pkg/mod/github.com/spf13/cobra@v0.0.4/command.go:800
github.com/k1LoW/tbls/cmd.Execute()
        /Users/craigrosie/github/tbls/cmd/root.go:59 +0x2e
main.main()
        /Users/craigrosie/github/tbls/main.go:32 +0x20
```

**Comment by k1LoW** [comment]
Hi @craigrosie ! Thank you !!

Could you try [this branch](https://github.com/k1LoW/tbls/pull/115) ?

**Comment by craigrosie** [comment]
Sure! Now getting:

```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x41f77d6]

goroutine 1 [running]:
database/sql.(*Rows).close(0x0, 0x0, 0x0, 0x0, 0x0)
        /usr/local/opt/go/libexec/src/database/sql/sql.go:2976 +0x66
database/sql.(*Rows).Close(0x0, 0x48c1dc2, 0x18a)
        /usr/local/opt/go/libexec/src/database/sql/sql.go:2972 +0x33
github.com/k1LoW/tbls/drivers/postgres.(*Postgres).Analyze(0xc00036b7f0, 0xc00026ac80, 0x498c960, 0xc00034fce0)
        /Users/craigrosie/github/tbls/drivers/postgres/postgres.go:279 +0x268d
github.com/k1LoW/tbls/datasource.Analyze(0x7ffeefbfe483, 0xa0, 0x0, 0x0, 0x0)
        /Users/craigrosie/github/tbls/datasource/datasource.go:74 +0x683
github.com/k1LoW/tbls/cmd.glob..func2(0x4e5abc0, 0xc000268ef0, 0x1, 0x1)
        /Users/craigrosie/github/tbls/cmd/doc.go:72 +0x111
github.com/spf13/cobra.(*Command).execute(0x4e5abc0, 0xc000268ec0, 0x1, 0x1, 0x4e5abc0, 0xc000268ec0)
        /Users/craigrosie/github/golang/pkg/mod/github.com/spf13/cobra@v0.0.4/command.go:766 +0x2ae
github.com/spf13/cobra.(*Command).ExecuteC(0x4e5a4a0, 0x0, 0x0, 0x45e9e5a)
        /Users/craigrosie/github/golang/pkg/mod/github.com/spf13/cobra@v0.0.4/command.go:850 +0x2fc
github.com/spf13/cobra.(*Command).Execute(...)
        /Users/craigrosie/github/golang/pkg/mod/github.com/spf13/cobra@v0.0.4/command.go:800
github.com/k1LoW/tbls/cmd.Execute()
        /Users/craigrosie/github/tbls/cmd/root.go:59 +0x2e
main.main()
        /Users/craigrosie/github/tbls/main.go:32 +0x20
```

**Comment by k1LoW** [comment]
@craigrosie Sorry 🙏 I made a simple mistake. Could you try once ?

**Comment by craigrosie** [comment]
@k1LoW I don't get the segault anymore, but now I get:
```
sql: Scan error on column index 0, name "view_definition": unsupported Scan, storing driver.Value type <nil> into type *string
```
If I run with `DEBUG=1 tbls ...` I get:
```
sql: Scan error on column index 0, name "view_definition": unsupported Scan, storing driver.Value type <nil> into type *string
github.com/k1LoW/tbls/drivers/postgres.(*Postgres).Analyze
        /Users/craigrosie/github/tbls/drivers/postgres/postgres.go:121
github.com/k1LoW/tbls/datasource.Analyze
        /Users/craigrosie/github/tbls/datasource/datasource.go:74
github.com/k1LoW/tbls/cmd.glob..func2
        /Users/craigrosie/github/tbls/cmd/doc.go:72
github.com/spf13/cobra.(*Command).execute
        /Users/craigrosie/github/golang/pkg/mod/github.com/spf13/cobra@v0.0.4/command.go:766
github.com/spf13/cobra.(*Command).ExecuteC
        /Users/craigrosie/github/golang/pkg/mod/github.com/spf13/cobra@v0.0.4/command.go:850
github.com/spf13/cobra.(*Command).Execute
        /Users/craigrosie/github/golang/pkg/mod/github.com/spf13/cobra@v0.0.4/command.go:800
github.com/k1LoW/tbls/cmd.Execute
        /Users/craigrosie/github/tbls/cmd/root.go:59
main.main
        /Users/craigrosie/github/tbls/main.go:32
runtime.main
        /usr/local/opt/go/libexec/src/runtime/proc.go:200
runtime.goexit
        /usr/local/opt/go/libexec/src/runtime/asm_amd64.s:1337
```

**Comment by k1LoW** [comment]
> sql: Scan error on column index 0, name "view_definition": unsupported Scan, storing driver.Value type <nil> into type *string

Oh.. Thanks !

**Comment by k1LoW** [comment]
@craigrosie 

I fixed https://github.com/k1LoW/tbls/issues/111#issuecomment-496206524 .

**Comment by craigrosie** [comment]
It works! 🎉 

Thanks again for your quick response!

**Comment by k1LoW** [comment]
@craigrosie Thank you for your support !!!!!

**Comment by k1LoW** [comment]
Released as v1.15.3.
